core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -137,6 +137,14 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
    * }}}
    */
   def toNested: Nested[F, Option, A] = Nested(value)
+
+  /**
+   * This repeats an F until we get defined values. This can be useful
+   * for polling type operations on State (or RNG) Monads, or in effect
+   * monads.
+   */
+  def untilDefined(implicit F: FlatMap[F]): F[A] =
```

Comment:
I didn't name this the same as `untilDefinedM` because it is slightly different. `OptionT[F, Option[A]]` should still have access to untilDefinedM. This can possibly be removed since it adds little over `optt.value.untilDefinedM`